pandas-stubs/_libs/tslibs/timedeltas.pyi

```diff
@@ -121,44 +132,169 @@ class Timedelta(timedelta):
     def ceil(self: _S, freq: str | BaseOffset) -> _S: ...
     @property
     def resolution_string(self) -> str: ...
-    def __add__(self, other: timedelta) -> Timedelta: ...
+    @overload  # type: ignore[override]
+    def __add__(self, other: timedelta | np.timedelta64) -> Timedelta: ...
+    @overload
+    def __add__(self, other: dt.datetime | np.datetime64 | Timestamp) -> Timestamp: ...
+    @overload
+    def __add__(self, other: Period) -> Period: ...
+    @overload
+    def __add__(self, other: dt.date) -> dt.date: ...
+    @overload
+    def __add__(self, other: PeriodIndex) -> PeriodIndex: ...
+    @overload
+    def __add__(self, other: DatetimeIndex) -> DatetimeIndex: ...
+    @overload
+    def __add__(
+        self, other: npt.NDArray[np.timedelta64]
+    ) -> npt.NDArray[np.timedelta64]: ...
+    @overload
+    def __add__(
+        self, other: npt.NDArray[np.datetime64]
+    ) -> npt.NDArray[np.datetime64]: ...
+    @overload
     def __radd__(self, other: timedelta) -> Timedelta: ...
-    def __sub__(self, other: timedelta) -> Timedelta: ...
-    def __rsub__(self, other: timedelta) -> Timedelta: ...
+    @overload
+    def __radd__(
+        self, other: npt.NDArray[np.timedelta64]
+    ) -> npt.NDArray[np.timedelta64]: ...
+    @overload
+    def __radd__(
+        self, other: npt.NDArray[np.datetime64]
+    ) -> npt.NDArray[np.datetime64]: ...
+    @overload  # type: ignore[override]
+    def __sub__(self, other: timedelta | np.timedelta64) -> Timedelta: ...
+    @overload
+    def __sub__(
+        self, other: npt.NDArray[np.timedelta64]
+    ) -> npt.NDArray[np.timedelta64]: ...
+    @overload
+    def __rsub__(self, other: timedelta | np.timedelta64) -> Timedelta: ...
+    @overload
+    def __rsub__(self, other: Timestamp | np.datetime64) -> Timestamp: ...
+    @overload
+    def __rsub__(self, other: PeriodIndex) -> PeriodIndex: ...
+    @overload
+    def __rsub__(self, other: DatetimeIndex) -> DatetimeIndex: ...
```

Comment:
I think I wrote something somewhere, but can't see it.  

It is because it makes sense to subtrace a timedelta from something that is like a date, .e.g., today - 1 day = yesterday, but not the reverse,, e.g., 1 day - today = ???.

The actual implementation follows this as well, and date-like minus TImedelta raises.